### PR TITLE
fix(GH-3798): update spring-boot.version to 2.6.2

### PR DESCRIPTION
**Pull request by igdianov**
This PR updates `spring-boot.version` to 2.6.2 release.

- [x] Removed circular dependencies in several auto configurations because Spring Boot 2.6+ started prohibiting circular references by default: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.6-Release-Notes#circular-references-prohibited-by-default

Part of https://github.com/Activiti/Activiti/issues/3798

### CODE REVIEW

**Review by andrea-ligios — APPROVED** [review]
LGTM!

**Review by erdemedeiros — APPROVED** [review]
LGTM!

File: activiti-examples/activiti-api-web-example/src/main/java/org/activiti/examples/DemoApplicationConfiguration.java [hunk 1]
```diff
@@ -83,12 +88,10 @@ protected void configure(HttpSecurity http) throws Exception {
...
                 .and()
                 .httpBasic();
-
-
     }
 
     @Bean
-    public PasswordEncoder passwordEncoder() {
+    public static PasswordEncoder passwordEncoder() {
         return new BCryptPasswordEncoder();
```
Comment (gicappa) [inline activiti-examples/activiti-api-web-example/src/main/java/org/activiti/examples/DemoApplicationConfiguration.java]: IIRC adding a static would return a new `BCryptPasswordEncoder`  for every method call. Is this desired?

**Review by gicappa — APPROVED** [review]
LGTM!

File: activiti-examples/activiti-api-web-example/src/main/java/org/activiti/examples/DemoApplicationConfiguration.java (hunk 1 above)
Comment (igdianov) [reply activiti-examples/activiti-api-web-example/src/main/java/org/activiti/examples/DemoApplicationConfiguration.java]: Spring will instantiate this bean as singleton only once using static method. This is done to break circular dependency between bean instance and configuration class instance that uses it.

**Pull request merged**